src/pages/home/Form.jsx

```diff
@@ -32,13 +40,31 @@ const Signup = () => {
 				}}
 				validationSchema={SignUpSchema}
 				onSubmit={(values) => {
+					/* Functionality inspired by https://github.com/gndx/react-mailchimp-form */
+					const url =
+						'https://uwbacm.us16.list-manage.com/subscribe/post?u=8773dffc3f528b0b18fd7aef8&amp;id=7a3e5f69db';
+					const data = Object.keys(values)
+						.map((key) => `${key}=${encodeURIComponent(values[key])}`)
+						.join('&');
+					const path = `${url}&${data}`;
+					const endpoint = path.replace('/post?', '/post-json?');
+					console.log('posting to the following endpoint: ' + endpoint);
 					/* Post to the survey form */
-					fetch('https://uwbacm.us16.list-manage.com/subscribe/post?u=8773dffc3f528b0b18fd7aef8&id=11b31196c4', {
-						method: 'post',
-						body: values,
-						mode: 'no-cors'
-					}).then((response) => {
+					jsonp(endpoint, {param: 'c'}, (err, response) => {
+						if (response.msg.includes('already subscribed')) {
+							/* do something better here */
+							userMessage = submittedMessages.duplicate;
```

Comment:
Mutating this variable isn’t going to cause a re-render, so if it’s being accessed in render then it should be in state.

Reply:
OK, the `useState()` declarations make sense now. Thanks!!! :grin: